subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -782,13 +782,15 @@ uint8_t ull_adv_aux_hdr_set_clear(struct ll_adv_set *adv,
 
 			pri_adi_prev = (void *)pri_dptr_prev;
 			did = sys_le16_to_cpu(pri_adi_prev->did);
+			did++;
 		} else {
+			static uint16_t did_unique;
+
 			pri_adi->sid = adv->sid;
 			sec_adi->sid = adv->sid;
+			did = did_unique++;
```

Comment:
How does this ensure uniqueness? Looks to me like it is still possible that the "old" `did` value and this `did_unique` value can still end up as the same value, as they are incremented separately. 

e.g. first iteration `did_unique is 0` (statically declared) and `did` is set to 0. Next iteration if `pri_hdr_prev.adi` is true, `did` is incremented to 1. If the next iteration has `pri_hdr_prev.adi` evaluated to false, then `did` is set to 1 again. 

Furthermore, this also doesn't seem to take this requirement into account: 

> The Advertising DID for a given advertising set or periodic advertising train shall be initialized with a randomly chosen value.

Reply:
`did_unique` value is used and post incremented only when an advertising set is created, i.e. Advertising Data Information is added to the extended common header of the PDU. ADI is not removed from the advertising until the set is removed/cleared. 
Hence, for a given set (SID) the DID in the PDU is privately incremented when the PDU is updated, and for new sets unique values from `did_unique` get used.
But yes, an advertising set starting with DID as `0` from `did_unique` after an update would have `1`, and if removed, and added back would get `1` from  `did_unique`.

Maybe a simple running value (initially a random number at power up) across all sets would avoid repeated `DID` ?

Reply:
> Maybe a simple running value (initially a random number at power up) across all sets would avoid repeated DID ?

That sounds like a solution that would ensure DID uniqueness as well as conforming to the other requirement (which I guess cannot be tested by any conformance test).